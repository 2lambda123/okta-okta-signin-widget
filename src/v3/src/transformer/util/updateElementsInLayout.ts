/*
 * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import {
  StepperLayout,
  UISchemaElement,
  UISchemaLayout,
  UISchemaLayoutType,
} from '../../types';

type PredicateFn = (uischema: UISchemaElement) => boolean;
type MapFn = (uischema: UISchemaElement) => UISchemaElement;
type CallbackFn = (uischema: UISchemaElement) => void;

type Options = {
  layout: UISchemaLayout;
  predicate: PredicateFn;
  callback: CallbackFn;
};

type OptionsOld = {
  layout: UISchemaLayout;
  mapFn: MapFn;
  predicateFn: PredicateFn;
};

export const updateElementsInLayout = (options: OptionsOld) => {
  const fn = (
    layout: UISchemaLayout,
    mapFn: MapFn,
    predicateFn: PredicateFn = () => true,
  ): UISchemaLayout => {
    // eslint-disable-next-line no-param-reassign
    layout.elements = layout.elements.map((element) => {
      const { type } = element;
      if (type === UISchemaLayoutType.STEPPER) {
        // eslint-disable-next-line no-param-reassign
        (element as StepperLayout).elements = (element as StepperLayout).elements
          .map((el) => fn(el, mapFn, predicateFn));
        return element;
      }

      if ([UISchemaLayoutType.HORIZONTAL, UISchemaLayoutType.VERTICAL]
        .includes(type as UISchemaLayoutType)) {
        return fn(element as UISchemaLayout, mapFn, predicateFn);
      }

      if (predicateFn(element)) {
        return mapFn(element);
      }

      return element;
    });

    return layout;
  };

  return fn(options.layout, options.mapFn, options.predicateFn);
};

export const traveseLayout = (options: Options) => {
  const fn = (
    layout: UISchemaLayout,
    predicateFn: PredicateFn,
    callback: CallbackFn,
  ) => {
    layout.elements.forEach((element) => {
      if (predicateFn(element)) {
        callback(element);
      }

      const { type } = element;

      if (type === UISchemaLayoutType.STEPPER) {
        (element as StepperLayout).elements
          .forEach((el) => fn(el, predicateFn, callback));
        return;
      }

      if ([UISchemaLayoutType.HORIZONTAL, UISchemaLayoutType.VERTICAL]
        .includes(type as UISchemaLayoutType)) {
        fn(element as UISchemaLayout, predicateFn, callback);
        return;
      }
    });
  };

  fn(options.layout, options.predicate, options.callback);
}
