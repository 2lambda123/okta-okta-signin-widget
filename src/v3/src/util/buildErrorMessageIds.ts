/*
 * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

export const buildErrorMessageIds = (errors: string[], fieldName: string): string => {
  if (errors.length === 1) {
    return `${fieldName}-error`;
  }
  const ids = errors.map((_: string, index: number) => `${fieldName}-error-${index}`);
  return ids.join(' ');
};
