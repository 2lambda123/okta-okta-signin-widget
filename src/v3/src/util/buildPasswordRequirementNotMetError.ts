/*
 * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import { ListItem, PasswordValidation } from '../types';
import { loc } from './locUtil';

export const buildPasswordRequirementNotMetError = (
  passwordRequirements: ListItem[],
  passwordValidations: PasswordValidation,
): string | undefined => {
  // TODO: Add ticket here to create new translation key
  const errorMessagePrefix = loc('registration.error.password.passwordRequirementsNotMet.prefix', 'login');
  const errorMessageSuffix = passwordRequirements.map((requirement: ListItem) => {
    const ruleValue = passwordValidations[requirement.ruleKey];
    return ruleValue === false ? requirement.label : undefined;
  }).filter(Boolean).join(', ') || undefined;

  return typeof errorMessageSuffix !== 'undefined'
    ? `${errorMessagePrefix} ${errorMessageSuffix}`
    : undefined;
};
