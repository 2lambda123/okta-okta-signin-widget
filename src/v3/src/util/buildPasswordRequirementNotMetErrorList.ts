/*
 * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import { IdxMessageWithName, ListItem, PasswordValidation } from '../types';
import { loc } from './locUtil';

export const buildPasswordRequirementNotMetErrorList = (
  passwordRequirements: ListItem[],
  passwordValidations: PasswordValidation,
  fieldName: string,
): IdxMessageWithName[] => {
  const errorMessages: IdxMessageWithName[] = [];

  passwordRequirements.forEach((requirement: ListItem) => {
    const ruleValue = passwordValidations[requirement.ruleKey];
    if (ruleValue === false) {
      errorMessages.push({
        name: fieldName,
        class: 'ERROR',
        message: requirement.label,
        i18n: { key: '' },
      });
    }
  });

  if (errorMessages.length > 0) {
    // TODO: Add ticket here to create new translation key that includes colon
    errorMessages.unshift({
      name: fieldName,
      class: 'ERROR',
      message: loc('registration.error.password.passwordRequirementsNotMet', 'login'),
      i18n: { key: 'registration.error.password.passwordRequirementsNotMet' },
    });
  }

  return errorMessages;
};
