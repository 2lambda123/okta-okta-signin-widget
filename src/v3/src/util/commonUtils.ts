/*
 * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

// eslint-disable-next-line
export const getDisplayName = (WrappedComponent: any): string => (
  WrappedComponent.displayName || WrappedComponent.name || 'Component'
);

export const handleFormFieldChange = (
  setDirtyFn: ((isDirty: boolean) => void) | undefined,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  handleChange: (path: string, value: any) => void,
  path: string,
  // eslint-disable-next-line
  value: any,
): void => {
  setDirtyFn?.(true);
  handleChange(path, value);
};
