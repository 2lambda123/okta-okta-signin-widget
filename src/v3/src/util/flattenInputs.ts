/*
 * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import { Input } from '@okta/okta-auth-js';

export const flattenInputs = (input: Input, name = ''): Input[] => {
  const res: Input[] = [];
  const { value } = input;

  if (Array.isArray(value)) {
    return value.reduce((acc, curr) => [
      ...acc,
      ...flattenInputs(curr, input.name),
    ], res);
  }

  return [{
    ...input,
    name: name ? `${name}.${input.name}` : input.name,
  }];
};
