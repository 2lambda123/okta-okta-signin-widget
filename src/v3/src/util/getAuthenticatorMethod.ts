/*
 * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import { IdxTransaction } from '@okta/okta-auth-js';

export const getAuthenticatorMethod = (
  transaction: IdxTransaction,
): string | undefined => {
  // currentAuthenticator is from enrollment flows and currentAuthenticatorEnrollment is from verify flows
  // @ts-ignore Add currentAuthenticatorEnrollment to IdxContext type
  const { context: { currentAuthenticator, currentAuthenticatorEnrollment } } = transaction;
  const enrollMethods = currentAuthenticator?.value?.methods;
  const verifyMethods = currentAuthenticatorEnrollment?.value?.methods;

  return (enrollMethods && enrollMethods[0].type) || (verifyMethods && verifyMethods[0].type);
};
