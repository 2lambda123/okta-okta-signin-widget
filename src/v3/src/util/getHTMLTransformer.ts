/*
 * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import { convertNodeToElement, Options } from 'react-html-parser';

export const getHTMLTransformer = (
  currentNodeName: string,
  newNodeName: string,
  attributes?: Record<string, unknown>,
): Options['transform'] => {
  const transform: Options['transform'] = (node, index) => {
    if (node.type === 'tag' && node.name === currentNodeName) {
      // eslint-disable-next-line no-param-reassign
      node.name = newNodeName;
      if (typeof attributes !== 'undefined') {
        // eslint-disable-next-line no-param-reassign
        node.attribs = attributes;
      }
      return convertNodeToElement(node, index, transform!);
    }
    return undefined;
  };

  return transform;
};
