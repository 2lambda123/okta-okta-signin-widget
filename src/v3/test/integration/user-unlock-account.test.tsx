/*
 * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
 * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
 *
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and limitations under the License.
 */

import { within } from '@testing-library/preact';
import { setup } from './util';

import mockResponse from '../../src/mocks/response/idp/idx/unlock-account/default.json';

describe('user-unlock-account', () => {
  it('renders form', async () => {
    const { container, findByText, findByLabelText } = await setup({ mockResponse });
    await findByText(/Unlock account\?/);
    await findByLabelText(/Username/);
    expect(container).toMatchSnapshot();
  });

  it('should display client-side validation errors when trying to submit the flow without a username', async () => {
    const {
      authClient, user, findByTestId, findByRole,
    } = await setup({ mockResponse });

    await findByTestId('identifier') as HTMLInputElement;
    const emailAuthenticatorButton = await findByTestId('okta_email');

    await user.click(emailAuthenticatorButton);
    expect(authClient.options.httpRequestClient).not.toHaveBeenCalledWith(
      'POST',
      'https://oie-4695462.oktapreview.com/idp/idx/challenge',
      expect.anything(),
    );
    const alertBox = await findByRole('alert');
    within(alertBox).findByText(/We found some errors/);
    const identifierError = await findByTestId('identifier-error');
    expect(identifierError.textContent).toEqual('This field cannot be left blank');
  });

  describe('send corrent payload', () => {
    it('when select email authenticator', async () => {
      const { authClient, user, findByTestId } = await setup({ mockResponse });

      const usernameEl = await findByTestId('identifier') as HTMLInputElement;
      await user.type(usernameEl, 'testuser@okta.com');
      expect(usernameEl.value).toEqual('testuser@okta.com');
      const emailAuthenticatorButton = await findByTestId('okta_email');
      await user.click(emailAuthenticatorButton);
      expect(authClient.options.httpRequestClient).toHaveBeenCalledWith(
        'POST',
        'https://oie-4695462.oktapreview.com/idp/idx/challenge',
        {
          data: JSON.stringify({
            identifier: 'testuser@okta.com',
            authenticator: {
              id: 'aut2h3fft0FbVvFYV1d7',
            },
            stateHandle: 'fake-stateHandle',
          }),
          headers: {
            Accept: 'application/json; okta-version=1.0.0',
            'Content-Type': 'application/json',
            'X-Okta-User-Agent-Extended': 'okta-auth-js/9.9.9',
          },
          withCredentials: true,
        },
      );
    });

    it('when select phone authenticator', async () => {
      const { authClient, user, findByTestId } = await setup({ mockResponse });

      const usernameEl = await findByTestId('identifier') as HTMLInputElement;
      await user.type(usernameEl, 'testuser@okta.com');
      expect(usernameEl.value).toEqual('testuser@okta.com');
      const phoneAuthenticatorButton = await findByTestId('phone_number');
      await user.click(phoneAuthenticatorButton);
      expect(authClient.options.httpRequestClient).toHaveBeenCalledWith(
        'POST',
        'https://oie-4695462.oktapreview.com/idp/idx/challenge',
        {
          data: JSON.stringify({
            identifier: 'testuser@okta.com',
            authenticator: {
              id: 'aut2h3fft10VeLDPD1d7',
            },
            stateHandle: 'fake-stateHandle',
          }),
          headers: {
            Accept: 'application/json; okta-version=1.0.0',
            'Content-Type': 'application/json',
            'X-Okta-User-Agent-Extended': 'okta-auth-js/9.9.9',
          },
          withCredentials: true,
        },
      );
    });

    it('when click "Back to sign in" button', async () => {
      const { authClient, user, findByTestId } = await setup({ mockResponse });

      const cancelButton = await findByTestId('cancel');
      await user.click(cancelButton);
      expect(authClient.options.httpRequestClient).toHaveBeenCalledWith(
        'POST',
        'https://oie-4695462.oktapreview.com/idp/idx/cancel',
        {
          data: JSON.stringify({
            stateHandle: 'fake-stateHandle',
          }),
          headers: {
            Accept: 'application/json; okta-version=1.0.0',
            'Content-Type': 'application/json',
            'X-Okta-User-Agent-Extended': 'okta-auth-js/9.9.9',
          },
          withCredentials: true,
        },
      );
    });
  });
});
