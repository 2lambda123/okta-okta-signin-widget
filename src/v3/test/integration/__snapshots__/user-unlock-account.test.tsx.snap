// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user-unlock-account renders form 1`] = `
<div>
  <div
    class="MuiScopedCssBaseline-root emotion-0"
  >
    <span>
      <main
        class="auth-container main-container mainViewContainer MuiBox-root emotion-1"
        data-commit="b9bbc0140703c3fbf0e2e58920362e70"
        data-version="0.0.0"
        id="okta-sign-in"
      >
        <div
          class="siwContainer MuiBox-root emotion-2"
        >
          <div
            class="okta-sign-in-header auth-header siwHeader"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 emotion-3"
            />
          </div>
          <div
            class="MuiBox-root emotion-4"
          >
            <form
              class="o-form"
              data-se="form"
              novalidate=""
              style="max-width: 100%;"
            >
              <div
                class="MuiBox-root emotion-5"
              >
                <div
                  class="MuiBox-root emotion-6"
                >
                  <div
                    class="MuiBox-root emotion-7"
                  >
                    <h2
                      class="MuiTypography-root MuiTypography-h3 emotion-8"
                      data-se="o-form-head"
                    >
                      Unlock account?
                    </h2>
                  </div>
                </div>
                <div
                  class="MuiBox-root emotion-6"
                >
                  <div
                    class="MuiBox-root emotion-10"
                  >
                    <label
                      class="MuiInputLabel-root MuiFormLabel-root MuiFormLabel-colorPrimary emotion-11"
                      for="identifier"
                    >
                      Username
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth emotion-12"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="username"
                        class="MuiOutlinedInput-input MuiInputBase-input emotion-13"
                        data-se="identifier"
                        id="identifier"
                        name="identifier"
                        type="text"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline emotion-14"
                      >
                        <legend
                          class="emotion-15"
                        >
                          <span
                            class="notranslate"
                          >
                            â€‹
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root emotion-6"
                >
                  <button
                    class="authButton MuiBox-root emotion-17"
                    data-se="authenticator-button"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="MuiBox-root emotion-10"
                      data-se="authenticator-icon"
                    >
                      <div
                        class="iconContainer mfa-okta-email MuiBox-root emotion-10"
                      >
                        <svg
                          aria-labelledby="okta_email_0"
                          fill="none"
                          height="48"
                          role="img"
                          viewBox="0 0 48 48"
                          width="48"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="okta_email_0"
                          >
                            Email Authentication
                          </title>
                          <circle
                            class="siwFillBg"
                            cx="24"
                            cy="24"
                            fill="#F5F5F6"
                            r="24"
                          />
                          <path
                            class="siwFillSecondary"
                            d="M32 18v1h5.3L32 24.29v1.42l6-6.01v12.8a1.5 1.5 0 0 1-1.5 1.5h-17a1.5 1.5 0 0 1-1.5-1.5V31h-1v1.5a2.5 2.5 0 0 0 2.5 2.5h17a2.5 2.5 0 0 0 2.5-2.5V18h-7Z"
                            fill="#A7B5EC"
                          />
                          <path
                            class="siwFillPrimary"
                            d="M9 13v14.5a2.5 2.5 0 0 0 2.5 2.5h17a2.5 2.5 0 0 0 2.5-2.5V13H9Zm20.293 1-8.81 8.81L10.794 14h18.499ZM28.5 29h-17a1.5 1.5 0 0 1-1.5-1.5V14.63l10.517 9.56L30 14.707V27.5a1.5 1.5 0 0 1-1.5 1.5Z"
                            fill="#00297A"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      class="infoSection MuiBox-root emotion-10"
                    >
                      <h3
                        class="MuiTypography-root MuiTypography-h3 emotion-21"
                        data-se="authenticator-button-label"
                      >
                        Email
                      </h3>
                      <div
                        class="cta-button actionName MuiBox-root emotion-10"
                        data-se="okta_email"
                      >
                        <span
                          class="MuiBox-root emotion-23"
                          data-se="cta-button-label"
                        >
                          Select
                        </span>
                        <mocksvgicon />
                      </div>
                    </div>
                  </button>
                </div>
                <div
                  class="MuiBox-root emotion-6"
                >
                  <button
                    class="authButton MuiBox-root emotion-17"
                    data-se="authenticator-button"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      class="MuiBox-root emotion-10"
                      data-se="authenticator-icon"
                    >
                      <div
                        class="iconContainer mfa-okta-phone MuiBox-root emotion-10"
                      >
                        <svg
                          aria-labelledby="phone_number_1"
                          fill="none"
                          height="48"
                          role="img"
                          viewBox="0 0 48 48"
                          width="48"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <title
                            id="phone_number_1"
                          >
                            Voice Call Authentication
                          </title>
                          <circle
                            class="siwFillBg"
                            cx="24"
                            cy="24"
                            fill="#F5F5F6"
                            r="24"
                          />
                          <path
                            class="siwFillPrimary"
                            d="M21.5 37h-9a2.6 2.6 0 0 1-2.5-2.682V14.682A2.6 2.6 0 0 1 12.5 12h9a2.6 2.6 0 0 1 2.5 2.682v19.636A2.6 2.6 0 0 1 21.5 37Zm-9-24a1.6 1.6 0 0 0-1.5 1.682v19.636A1.6 1.6 0 0 0 12.5 36h9a1.6 1.6 0 0 0 1.5-1.682V14.682A1.6 1.6 0 0 0 21.5 13h-9Z"
                            fill="#00297A"
                          />
                          <path
                            class="siwFillPrimary"
                            d="M19 34h-4v1h4v-1Zm-1-19a1 1 0 1 0-2 0 1 1 0 0 0 2 0Z"
                            fill="#00297A"
                          />
                          <path
                            class="siwFillSecondary"
                            d="M34.606 34.606 33.9 33.9a13.998 13.998 0 0 0 0-19.8l.707-.707a15 15 0 0 1 0 21.212l-.001.001Z"
                            fill="#A7B5EC"
                          />
                          <path
                            class="siwFillSecondary"
                            d="m31.071 31.071-.707-.707a9 9 0 0 0 0-12.728l.707-.707a10 10 0 0 1 0 14.142Zm-3.535-3.536-.707-.707a4 4 0 0 0 0-5.656l.707-.707a5 5 0 0 1 0 7.07Z"
                            fill="#A7B5EC"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      class="infoSection MuiBox-root emotion-10"
                    >
                      <h3
                        class="MuiTypography-root MuiTypography-h3 emotion-21"
                        data-se="authenticator-button-label"
                      >
                        Phone
                      </h3>
                      <div
                        class="cta-button actionName MuiBox-root emotion-10"
                        data-se="phone_number"
                      >
                        <span
                          class="MuiBox-root emotion-23"
                          data-se="cta-button-label"
                        >
                          Select
                        </span>
                        <mocksvgicon />
                      </div>
                    </div>
                  </button>
                </div>
                <div
                  class="MuiBox-root emotion-6"
                >
                  <a
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-33"
                    data-se="cancel"
                    href="javascript:void(0)"
                  >
                    Back to sign in
                  </a>
                </div>
              </div>
            </form>
          </div>
        </div>
      </main>
    </span>
  </div>
</div>
`;
